src/databricks/labs/lsql/dashboards.py

```diff
@@ -728,6 +766,118 @@ def _get_query_widget_spec(fields: list[Field], *, frame: WidgetFrameSpec | None
         return spec
 
 
+@dataclass
+class FilterTile(Tile):
+    _position: Position = dataclasses.field(default_factory=lambda: Position(0, 0, 3, 2))
+
+    def validate(self) -> None:
+        """Validate the tile
+
+        Raises:
+            ValueError : If the tile is invalid.
+        """
+        if not self.metadata.is_filter():
+            raise ValueError(f"Tile is not a filter file: {self}")
+        if len(self.metadata.filters) == 0:
+            raise ValueError(f"Filter tile with path {self.metadata.path} has no filters defined")
+
+    def get_layouts(self, datasets: list[Dataset]) -> Iterable[Layout]:
+        """Get the layout(s) reflecting this tile in the dashboard."""
+        dataset_columns = self._get_dataset_columns(datasets)
+        if len(dataset_columns) == 0:
+            err_msg = f"Filter tile with path {self.metadata.path} has no matching dataset columns"
+            raise ValueError(err_msg)
+        widget = self._create_filter_widget(dataset_columns)
+        layout = Layout(widget=widget, position=self.position)
+        yield layout
+
+    def _create_filter_widget(self, dataset_columns: set[tuple[str, str]]) -> Widget:
+        filter_type = self.metadata.widget_type
+        match filter_type:
+            case WidgetType.MULTI_SELECT:
+                return self._create_multi_select_widget(dataset_columns)
+            case WidgetType.DATE_RANGE_PICKER:
+                return self._create_date_range_picker_widget(dataset_columns)
+            case _:
+                raise NotImplementedError(f"Filter type {filter_type} not supported for filter tile: {self}")
+
+    def _create_multi_select_widget(self, dataset_columns: set[tuple[str, str]]) -> Widget:
+        control_encodings: list[ControlEncoding] = []
+        queries = []
+        for idx, (column, dataset_name) in enumerate(dataset_columns):
+            filter_fields = [
+                Field(name=column, expression=f"`{column}`"),
+                Field(name=f"{column}_associativity", expression="COUNT_IF(`associative_filter_predicate_group`)"),
+            ]
+            filter_query = Query(dataset_name=dataset_name, fields=filter_fields, disaggregated=False)
+            filter_named_query = NamedQuery(name=f"multi_select_{column}_{idx}", query=filter_query)
```

Comment:
Is the `idx` required?

Reply:
Same column name can exist in multiple datasets. idx will make the NamedQueries unique.

Reply:
use dataset name instead of idx

Reply:
Done